api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -266,7 +266,7 @@ type LoadBalancerProfile struct {
 type APIServerAccessProfile struct {
 	// AuthorizedIPRanges - Authorized IP Ranges to kubernetes API server.
 	// +optional
-	AuthorizedIPRanges []string `json:"authorizedIPRanges,omitempty"`
+	AuthorizedIPRanges *[]string `json:"authorizedIPRanges,omitempty"`
```

Comment:
why use a pointer to a slice here? is an empty slice not enough info to tell us this should be empty?

Reply:
While an empty slice gives us enough information about it being empty, changing `AuthorizedIPRanges` to a `*[]string` type makes it consistent with Azure SDK's `AuthorizedIPRanges` data type.

Reply:
A minor improvement is that we no longer need to check the length of the array to determine if `AuthorizedIPRanges` is empty. But I don't think checking the length of `AuthorizedIPRanges` is really a big check.

Reply:
Updated the data type to `[]string` instead of a pointer.